app/models/service_ansible_playbook.rb

```diff
@@ -14,6 +14,25 @@ def preprocess(action, add_options = {})
   end
 
   def execute(action)
+    task_opts = {
+      :action => "Executing Ansible Playbook",
+      :userid => "system"
+    }
+
+    queue_opts = {
+      :args        => [action],
+      :class_name  => self.class.name,
+      :instance_id => id,
+      :method_name => "_execute",
+      :role        => "embedded_ansible"
+    }
+
+    task_id = MiqTask.generic_action_with_callback(task_opts, queue_opts)
```

Comment:
I don't think this is worth holding up the merge of this PR, but this looks very similar to what is available through `EmbeddedAnsible::CrudCommon`:

https://github.com/ManageIQ/manageiq/blob/82fe63826beb4fa43a472fc93d7ee3fd9e800e6e/app/models/manageiq/providers/embedded_ansible/crud_common.rb#L5-L21

So it would nice if there was a clean way to share this code without having to dig into the provider code to do it.  But again, I think this is fine as a surgical fix and I think trying to DRY this up can be done at a later date.

Reply:
That said, I guess my question for this new implementation is what is now happening with this line (which exists in `_execute` now:

https://github.com/ManageIQ/manageiq/blob/82fe63826beb4fa43a472fc93d7ee3fd9e800e6e/app/models/service_ansible_playbook.rb#L28

Is that just going to cause the same issue that we are seeing now, just with another queue item in between it?

(Note:  You can just expand the code in the diff below to see what I talking about, just shared a permalink since I can't comment on code that hasn't changed)

Reply:
So I don't think it's really worth DRYing this particular bit up. I think this is kinda how the task/queue stuff is meant to be used. Plus I think there's a fair chance that the args here could change separately from the provider CRUD stuff.

I was thinking that there could be a use case for creating an MiqTask helper to do the wait_for_taskid and raise if the status is no-good, but we're not doing that in crud_common so ... shrug.

Reply:
> Is that just going to cause the same issue that we are seeing now, just with another queue item in between it?

No, at this point in the thread of execution we will queue new items for states in the job, but those are all queued for the same server here: https://github.com/ManageIQ/manageiq/blob/82fe63826beb4fa43a472fc93d7ee3fd9e800e6e/app/models/manageiq/providers/ansible_runner_workflow.rb#L62-L78

Reply:
Ah, sorry, I see now.  It is a bit farther down then the preview showed:

https://github.com/ManageIQ/manageiq/blob/82fe63826beb4fa43a472fc93d7ee3fd9e800e6e/app/models/manageiq/providers/ansible_runner_workflow.rb#L76

I was seeing the `*args` and was trying to find where it was being passed in from 😩 

Okay, cool, looks good to me.